src/arithmetic/algebraic_expression.c

```diff
@@ -371,84 +380,60 @@ AlgebraicExpression **AlgebraicExpression_From_Query(const AST *ast, Vector *mat
     exp->dest_node = dest;
     AlgebraicExpression **expressions = _AlgebraicExpression_Intermidate_Expressions(exp, ast, matchPattern, q, exp_count);
     expressions = _AlgebraicExpression_IsolateVariableLenExps(expressions, exp_count);
-    AlgebraicExpression_Free(exp);
+    // AlgebraicExpression_Free(exp);
+    free(exp);
```

Comment:
Running the query:
`GRAPH.QUERY social "MATCH (a)-[e:friend|:visited|:new]->(b) RETURN a, b, e"`
`expressions` and `exp->operands` do not get freed.

Reply:
:+1: